### PR TITLE
remove dupe method, fix func call, fix timeout parse

### PR DESCRIPTION
**Pull request by gravitypriest**
The registration refactoring brought about some bugs we didn't catch til today, namely:

- still using `InsightsClient.get_connection()`
- `http_timeout` not being parsed to a float correctly

This commit fixes those, as well as removing the duplicate function `get_registration_information` in favor of `get_registration_status`, to keep name parity with the matching function in `client.py`.

### CODE REVIEW

**Comment by gravitypriest** [comment]
That function's still used. `registration_check` is where all the response parsing is done.

File: insights/client/config.py [hunk 1]
```diff
@@ -487,10 +487,20 @@ def load_config_file(self, fname=None):
...
+                    d[key] = parsedconfig.getfloat(constants.app_name, key)
+                if key in DEFAULT_BOOLS and isinstance(
+                        d[key], six.string_types):
+                    d[key] = parsedconfig.getboolean(constants.app_name, key)
+            except ValueError as e:
+                if self._print_errors:
+                    sys.stdout.write(
+                        'ERROR: {0}.\nCould not read configuration file, '
+                        'using defaults\n'.format(e))
+                return
```
Comment (kylape) [inline insights/client/config.py]: So returning here means it just stops processing the remaining keys?  Seems like it should either halt the whole process or keep going.

Comment (gravitypriest) [reply insights/client/config.py]: The config object isn't updated with the values from the .conf file until the very end of the function. So if any exception forces a return it won't be changed from the defaults. We've never supported a case where we only parse _some_ of the config vars from the file.

Comment (kylape) [reply insights/client/config.py]: Makes sense, thanks.

**Pull request merged**